src/Discord/Parts/User/Member.php

```diff
@@ -238,27 +247,27 @@ public function removeRole($role, ?string $reason = null): ExtendedPromiseInterf
             $role = $role->id;
         }
 
-        if (in_array($role, $this->attributes['roles'])) {
-            $headers = [];
-            if (isset($reason)) {
-                $headers['X-Audit-Log-Reason'] = $reason;
-            }
+        if (! in_array($role, $this->attributes['roles'])) {
+            return reject(new \RuntimeException('Member does not have role.'));
```

Comment:
from The API specification, it will error when you are trying to remove a non existent role, which then I assume developers would follow the docs and catch this kind of error.

Reply:
I can't find the info about error on non-existent role removal.

I still think it shouldn't be a `\RuntimeException`, since that kind of exceptions should only be thrown in case if the user cannot circumvent the situation where this exception gets thrown. 

People can check if member has a role before calling `removeRole()` to avoid the exception throw.

Reply:
No exception is thrown unless user explicitly set `->done()` on the promise and did not catch it.